tests/unit/base/test_HITParser1.i

```diff
@@ -2,6 +2,12 @@ x = 5
 pi = 3.14159
 day = 'day'
 
+[Settings]
+  default_floating_point_type = 'Float16'
+  default_integer_type = 'Int32'
+  default_device = 'cuda:1'
```

Comment:
What does this do if you don't have cuda?

Reply:
In `Settings.cxx` this is parsed as a `std::string` and converted to a `torch::Device`. `torch::Device` is always "valid" (as in it keeps track of the device type and device id), and errors are only triggered when someone tries to use the `torch::Device`, i.e. sending a tensor to that device.

So, this should be a valid test case for testing correctly _parsing_ the default device, and nothing more.

Reply:
Okay.